kv/bucket.go

```diff
@@ -119,52 +119,48 @@ func (s *Service) FindBucketByName(ctx context.Context, orgID influxdb.ID, n str
 	span, ctx := tracing.StartSpanFromContext(ctx)
 	defer span.Finish()
 
-	internal, err := s.findSystemBucket(n)
-	// if found in our internals list, return mock
-	if err == nil {
-		internal.OrgID = orgID
-		return internal, nil
-	}
-
 	var b *influxdb.Bucket
-
-	err = s.kv.View(ctx, func(tx Tx) error {
+	err := s.kv.View(ctx, func(tx Tx) error {
 		bkt, pe := s.findBucketByName(ctx, tx, orgID, n)
 		if pe != nil {
-			err = pe
+			err := pe
 			return err
 		}
+
 		b = bkt
 		return nil
 	})
 
 	return b, err
 }
 
-func (s *Service) findSystemBucket(n string) (*influxdb.Bucket, error) {
-	switch n {
-	case "_tasks":
-		return &influxdb.Bucket{
-			ID:              influxdb.TasksSystemBucketID,
-			Type:            influxdb.BucketTypeSystem,
-			Name:            "_tasks",
-			RetentionPeriod: time.Hour * 24 * 3,
-			Description:     "System bucket for task logs",
-		}, nil
-	case "_monitoring":
-		return &influxdb.Bucket{
-			ID:              influxdb.MonitoringSystemBucketID,
-			Type:            influxdb.BucketTypeSystem,
-			Name:            "_monitoring",
-			RetentionPeriod: time.Hour * 24 * 7,
-			Description:     "System bucket for monitoring logs",
-		}, nil
-	default:
-		return nil, &influxdb.Error{
-			Code: influxdb.ENotFound,
-			Msg:  fmt.Sprintf("system bucket %q not found", n),
-		}
+// CreateSystemBuckets creates the task and monitoring system buckets for an organization
+func (s *Service) createSystemBuckets(ctx context.Context, tx Tx, o *influxdb.Organization) error {
+	tb := &influxdb.Bucket{
+		OrgID:           o.ID,
+		Type:            influxdb.BucketTypeSystem,
+		Name:            influxdb.TasksSystemBucketName,
+		RetentionPeriod: time.Hour * 24 * 3,
```

Comment:
This retention period has popped up again: I think it should be a constant somewhere.